daemon/indexing/listener/listener.js

```diff
@@ -182,9 +174,40 @@ async function runBatch(opts, context) {
   return lastLogBlock
 }
 
+async function withRetrys(fn) {
+  let tryCount = 0
+  while (true) {
+    try {
+      await fn() // Do our action.
+      return // it worked!
+    } catch (e) {
+      // Roughly double wait time each failure
+      let waitTime = Math.pow(100, 1 + tryCount / 6)
+      // Randomly jiggle wait time by 20% either way. No thundering herd.
+      waitTime = Math.floor(waitTime * (1.2 - Math.random() * 0.4))
+      // Max out at two minutes
+      waitTime = Math.min(waitTime, 2 * 60 * 1000)
```

Comment:
nit: define maxRetryCount and maxWaitTime as constants so that if we decide to change them later on it's very easy to do so.